Bottom: bc63ea7fd6a39e18f813e5cc86541a362e44cffa
Top:    187a75796f68b2b09b3de87573aa911e15f11129
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-21 16:33:01 +0100

draw.c: Tidy up hole_callback routine


---

diff --git a/src/draw.c b/src/draw.c
index 8b9b86a..d6dce62 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -298,19 +298,11 @@ hole_callback (const BoxType * b, void *cl)
 {
   PinTypePtr pin = (PinTypePtr) b;
   int plated = cl ? *(int *) cl : -1;
-  switch (plated)
-    {
-    case -1:
-      break;
-    case 0:
-      if (!TEST_FLAG (HOLEFLAG, pin))
-	return 1;
-      break;
-    case 1:
-      if (TEST_FLAG (HOLEFLAG, pin))
-	return 1;
-      break;
-    }
+
+  if ((plated == 0 && !TEST_FLAG (HOLEFLAG, pin)) ||
+      (plated == 1 &&  TEST_FLAG (HOLEFLAG, pin)))
+    return 1;
+
   DrawHole ((PinTypePtr) b);
   return 1;
 }
